packages/odyssey-react-mui/src/MenuButton.tsx

```diff
@@ -45,13 +41,7 @@ export type MenuButtonProps = {
   /**
    * The <MenuItem> components within the Menu.
    */
-  children:
-    | ReactElement<typeof MenuItem | typeof Divider | typeof ListSubheader>
-    | Array<
-        | ReactElement<typeof MenuItem | typeof Divider | typeof ListSubheader>
-        | NullElement
-      >
-    | NullElement;
+  children: ReactNode | NullElement;
```

Comment:
@chrispulsinelli-okta What problem are we solving here exactly? Conditional rendering? Are we code with allowing anything to be rendered in the menu item list?

Reply:
A couple of things, actually.
From my usage of this, `ReactElement<typeof MenuItem | typeof Divider | typeof ListSubheader>` doesn't actually restrict one from using a `Button`, for example, where we think the generic typing restricts it.

Also, this doesn't allow for `null` or `undefined` to be passed in, which are valid JSX values to be passed to render.

`ReactNode` is a bit more lax in what it allows, allows for nullish values, and also arrays.

Reply:
I've mentioned this in another comment in this PR, but will address it here as well. 

> Part of the reason we used specific types for children is to ensure other devs know what we expect to be passed in.

Understood and I agree with the sentiments that we should be clear about what we expect our users to provide here, but this expression of the type is both hard to read and doesn't actually restrict users to provide the explicit types of react elements listed here.

Personally, I believe it's best to document the expectations in comments and stories and make it much easier to read the type.

>Using ReactNode (we shouldn't need NullElement in that case) opens it up so devs can pass anything and won't know exactly what we expect without looking at Storybook.

Using `ReactElement` also allows users to pass any component of generic type `ReactElement`. We actually are not preventing them from passing in any other components than the ones explicitly mentioned here.

Reply:
`ReactElement` is for React components, but I don't think it works for HTML.

`ReactNode` is any valid React component, HTML element, `null`, and `false` if I'm remembering correctly.

Reply:
> I'm a bit confused by your statement. I get that the TS doesn't work correctly, but you wrote ReactElement at the end, not ReactNode like the code.

Sorry for the confusion. I'm simply trying to make the point that using `ReactElement<typeof FooComponent>` doesn't guarantee us that users will provide and only provide a `FooComponent`.